lib/core/src/display/navigation/animation_manager.rs

```diff
@@ -0,0 +1,23 @@
+// ========================
+// === AnimationManager ===
+// ========================
+
+pub struct AnimationManager {
+    maximum_frame_duration: f32
+}
+
+impl AnimationManager {
+    pub fn new(minimum_fps : f32) -> Self {
+        let maximum_frame_duration = 1.0 / minimum_fps;
+        Self { maximum_frame_duration }
+    }
+
+    /// Runs the closure guaranteeing it runs at `minimum_fps`.
+    pub fn run<F:FnMut(f32)>(&mut self, mut dt:f32, mut f:F) {
```

Comment:
This is a public function which consumes a `dt` argument. What is `dt`? Is it diff of time? If so, why a public function consumes diff of time? Diffing of time should be internal to AnimationManager and user of this API should rather not provide it explicitly, right?